org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterBatteryAlarm.java

```diff
@@ -0,0 +1,166 @@
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.binding.zigbee.ZigBeeBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zsmartsystems.zigbee.CommandResult;
+import com.zsmartsystems.zigbee.ZigBeeEndpoint;
+import com.zsmartsystems.zigbee.zcl.ZclAttribute;
+import com.zsmartsystems.zigbee.zcl.ZclAttributeListener;
+import com.zsmartsystems.zigbee.zcl.clusters.ZclPowerConfigurationCluster;
+import com.zsmartsystems.zigbee.zcl.protocol.ZclClusterType;
+
+/**
+ * Converter for a battery alarm channel.
+ * <p>
+ * This converter relies on reports for the BatteryAlarmState attribute of the power configuration cluster, setting the
+ * state of the battery alarm channel depending on the bits set in the BatteryAlarmState.
+ * <p>
+ * Possible future improvements:
+ * <ul>
+ * <li>The BatteryAlarmState provides battery level information for up to three batteries; this converter only considers
+ * the information for the first battery.
+ * <li>Devices might use alarms from the Alarms cluster instead of the BatteryAlarmState attribute to indicate battery
+ * alarms. This is currently not supported by this converter.
+ * <li>Devices might permit to configure the four battery level/voltage thresholds on which battery alarms are signaled;
+ * such configuration is currently not supported.
+ * </ul>
+ *
+ * @author Henning Sudbrock - Initial Contribution
+ */
+public class ZigBeeConverterBatteryAlarm extends ZigBeeBaseChannelConverter implements ZclAttributeListener {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private static final int MIN_THRESHOLD_BITMASK = 0b0001;
+    private static final int THRESHOLD_1_BITMASK = 0b0010;
+    private static final int THRESHOLD_2_BITMASK = 0b0100;
+    private static final int THRESHOLD_3_BITMASK = 0b1000;
+
+    private static final int BATTERY_ALARM_POLLING_PERIOD = (int) ofMinutes(30).getSeconds();
+
+    private ZclPowerConfigurationCluster cluster;
+
+    @Override
+    public boolean initializeConverter() {
+        logger.debug("{}: Initialising device battery alarm converter", endpoint.getIeeeAddress());
+
+        cluster = (ZclPowerConfigurationCluster) endpoint.getInputCluster(ZclPowerConfigurationCluster.CLUSTER_ID);
+        if (cluster == null) {
+            logger.error("{}: Error opening power configuration cluster", endpoint.getIeeeAddress());
+            return false;
+        }
+
+        try {
+            CommandResult bindResponse = cluster.bind().get();
+            if (bindResponse.isSuccess()) {
+                CommandResult reportingResponse = cluster.setReporting(cluster.getAttribute(ATTR_BATTERYALARMSTATE),
+                        (int) ofMinutes(10).getSeconds(), (int) ofHours(2).getSeconds()).get();
```

Comment:
Shouldn't we put these magic numbers, "10" and "2" into constants?

Reply:
fine for me, will update